File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/commands/PaymentErrorReceivedCommand.java
package com.hedvig.paymentservice.domain.trustlyOrder.commands;

import java.util.UUID;

import com.hedvig.paymentService.trustly.data.response.Error;

import org.axonframework.commandhandling.TargetAggregateIdentifier;

import lombok.Value;

@Value
public class PaymentErrorReceivedCommand {
    @TargetAggregateIdentifier
    UUID hedvigOrderId;

    Error error;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/commands/PaymentErrorReceivedCommand.kt
package com.hedvig.paymentservice.domain.trustlyOrder.commands

import com.hedvig.paymentService.trustly.data.response.Error
import org.axonframework.commandhandling.TargetAggregateIdentifier
import java.util.*

data class PaymentErrorReceivedCommand(
  @TargetAggregateIdentifier
  val hedvigOrderId: UUID,
  val error: Error
)
